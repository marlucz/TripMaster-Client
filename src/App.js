import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';

import Trips from 'pages/Trips/Trips';
import NotFound from 'pages/NotFound/NotFound';
import Expenses from 'pages/Trip/Expenses/Expenses';
import Itinerary from 'pages/Trip/Itinerary/Itinerary';
import TodoList from 'pages/Trip/TodoList/TodoList';
import Register from 'pages/Auth/Register';
import Login from 'pages/Auth/Login';
import Forgot from 'pages/Auth/Forgot';

import GlobalTemplate from 'templates/GlobalTemplate';

const App = () => (
    <BrowserRouter>
        <GlobalTemplate>
            <Switch>
                <Route exact path="/" component={Login} />
                <Route path="/register" component={Register} />
                <Route path="/login" component={Login} />
                <Route path="/forgot" component={Forgot} />
                <Route exact path="/trips" component={Trips} />
                <Route path="/trip/:id/expenses" component={Expenses} />
                <Route path="/trip/:id/itinerary" component={Itinerary} />
                <Route path="/trip/:id/todo" component={TodoList} />
                <Route component={NotFound} />
            </Switch>
        </GlobalTemplate>
    </BrowserRouter>
);

export default App;
