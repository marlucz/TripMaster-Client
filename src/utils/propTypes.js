import PropTypes from 'prop-types';

export const ItineraryItemPropTypes = PropTypes.shape({
    _id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
    name: PropTypes.string.isRequired,
    startDate: PropTypes.oneOfType([
        PropTypes.instanceOf(Date),
        PropTypes.string,
    ]).isRequired,
    location: PropTypes.shape({
        address: PropTypes.string,
        coordinates: PropTypes.array,
    }).isRequired,
    description: PropTypes.string,
    status: PropTypes.string.isRequired,
});
