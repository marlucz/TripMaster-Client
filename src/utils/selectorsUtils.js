export const sortByKey = (array, key) => {
    return array.sort((a, b) => {
        const x = a[key];
        const y = b[key];
        // eslint-disable-next-line
        return x < y ? -1 : x > y ? 1 : 0;
    });
};
