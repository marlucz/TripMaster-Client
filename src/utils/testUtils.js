import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from 'store/store';

export const withRouter = Component => props => (
    <BrowserRouter>
        <Component {...props} />
    </BrowserRouter>
);

export const withProvider = Component => props => (
    <Provider store={store}>
        <BrowserRouter>
            <Component {...props} />
        </BrowserRouter>
    </Provider>
);
