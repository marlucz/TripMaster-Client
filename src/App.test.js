import { render, wait, waitForElement } from '@testing-library/react';
import { withProvider } from 'utils/testUtils';
import { _App } from './App';

const AppWithProvider = withProvider(_App);

describe('<App />', () => {
    const props = {
        isAuth: false,
    };

    // eslint-disable-next-line
    const setup = props => {
        const utils = render(AppWithProvider(props));
        return { ...utils };
    };

    it('renders <App /> component', async () => {
        const { asFragment } = setup(props);
        await wait(); // wait for import to be resolved

        expect(asFragment()).toMatchSnapshot();
    });

    it('renders <Spinner /> when `isAuth` prop is null', async () => {
        const { getByTestId } = setup({ ...props, isAuth: null });
        const spinner = await waitForElement(() => getByTestId('Spinner'));

        expect(spinner).toBeInTheDocument();
    });
});
