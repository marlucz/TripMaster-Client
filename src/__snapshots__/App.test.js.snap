// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders <App /> component 1`] = `
<DocumentFragment>
  <div
    class="sc-fzoLsD jyEnQr"
  >
    <div
      class="sc-fzpans sc-fzplWN crGuTY"
    />
    <div
      class="sc-fzpans sc-fznyAO bBeVwp"
    />
  </div>
  <div>
    <div
      class="sc-AxhCb msitV"
    >
      <h1
        class="sc-AxhUy biDTtC"
      >
        TripMaster
      </h1>
      <form
        action="#"
        class="sc-AxgMl cCduxA"
      >
        <div
          class="sc-AxheI dPFwAA"
        >
          <input
            class="sc-Axmtr UpGFG"
            name="email"
            placeholder="Login"
            type="email"
            value=""
          />
          <input
            class="sc-Axmtr UpGFG"
            name="password"
            placeholder="Password"
            type="password"
            value=""
          />
          <a
            class="sc-AxmLO hiPRJm"
            href="/auth/forgot"
          >
            Forgot you password?
          </a>
        </div>
        <div
          class="sc-fzozJi dwmVYI"
        >
          <button
            class="sc-AxiKw eMyAEm"
            type="submit"
          >
            Sign In
          </button>
          <a
            class="sc-AxmLO hiPRJm"
            href="/auth/register"
          >
            Don't have an account?
          </a>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
